drivers/serial/uart_xlnx_uartlite.c

```diff
@@ -37,6 +37,10 @@
 #define CTRL_REG_RST_RX_FIFO BIT(1)
 #define CTRL_REG_ENABLE_INTR BIT(4)
 
+/* spin locks for RX and TX FIFO preventing a bus error */
+struct k_spinlock rx_lock;
+struct k_spinlock tx_lock;
```

Comment:
Based on the explanation provided in the commit message, global locking does not seem to be the correct solution (i.e. locking needs to be done per device instance), in which case these locks need to be declared in the `xlnx_uartlite_data` struct.

Reply:
Thanks!
You are absolutely right that these spinlocks should be per device.

Fixed.